test/integration/framework/etcd.go

```diff
@@ -221,14 +222,19 @@ func checkGoroutines(before int) error {
 	// It generally takes visibly less than 1s to finish all goroutines.
 	// But we keep the limit higher to account for cpu-starved environments.
 	if err := wait.Poll(100*time.Millisecond, 5*time.Second, checkNumberOfGoroutines); err != nil {
-		after := runtime.NumGoroutine()
-		stacktraces := make([]byte, 1<<20)
-		runtime.Stack(stacktraces, true)
-		return fmt.Errorf("unexpected number of goroutines: before: %d after %d\n%sd", before, after, string(stacktraces))
+		after, stacktracesAfter := stacks(1 << 20)
```

Comment:
this was more an internal detail to help debugging to really report it, but it doesn't hurt to improve the output